webodf/lib/gui/TrivialUndoManager.js

```diff
@@ -37,32 +37,124 @@ gui.TrivialUndoManager = function TrivialUndoManager(defaultRules) {
         domUtils = core.DomUtils,
         /**@type{?Element}*/
         initialDoc,
-        /**@type{!Array.<!ops.Operation>}*/
-        initialState = [],
+        /**@type{!StateTransition}*/
+        initialStateTransition,
         playFunc,
         /**@type{!ops.Document}*/
         document,
-        /**@type{!Array.<!ops.Operation>}*/
-        currentUndoState = [],
-        /**@type{!Array.<!Array.<!ops.Operation>>}*/
-        undoStates = [],
-        /**@type{!Array.<!Array.<!ops.Operation>>}*/
-        redoStates = [],
+        /**@type {!StateId}*/
+        unmodifiedStateId,
+        /**@type{!StateTransition}*/
+        currentUndoStateTransition,
+        /**@type{!Array.<!StateTransition>}*/
+        undoStateTransitions = [],
+        /**@type{!Array.<!StateTransition>}*/
+        redoStateTransitions = [],
         eventNotifier = new core.EventNotifier([
             gui.UndoManager.signalUndoStackChanged,
             gui.UndoManager.signalUndoStateCreated,
             gui.UndoManager.signalUndoStateModified,
+            gui.UndoManager.signalDocumentModifiedChanged,
             gui.TrivialUndoManager.signalDocumentRootReplaced
         ]),
         undoRules = defaultRules || new gui.UndoStateRules(),
-        isExecutingOps = false;
+        isExecutingOps = false,
+        /**
+        * Base for generating unique state ids
+        * @type {!number}
+        */
+        stateIdBase = 0;
+
+    /**
+     * @constructor
+     * @param {!number=} mainId
+     * @param {!number=} subId
+     */
+    function StateId(mainId, subId) {
```

Comment:
In my opinion it would, simply because it reduces the exhaustion in trying to parse such a large class. But, I am willing to be ignored on this :-).